synth.py

```diff
@@ -22,60 +22,53 @@
 logging.basicConfig(level=logging.DEBUG)
 
 # Run the gapic generator
-gapic = gcp.GAPICGenerator()
+gapic = gcp.GAPICMicrogenerator()
 versions = ['v1beta1', 'v1']
 for version in versions:
-    library = gapic.node_library('containeranalysis', version,
-            config_path=f"/google/devtools/containeranalysis/artman_containeranalysis_{version}.yaml")
-    s.copy(library, excludes=['package.json', 'README.md', 'src/index.js'])
+    library = gapic.typescript_library(
+      'containeranalysis', version,
+      generator_args={
+            "grpc-service-config": f"google/devtools/containeranalysis/{version}/containeranalysis_grpc_service_config.json",
+            "package-name": f"@google-cloud/containeranalysis",
+            "main-service": f"containeranalysis"
+            },
+        proto_path=f'/google/devtools/containeranalysis/{version}',
+        extra_proto_files=["google/cloud/common_resources.proto", "grafeas/v1"]
+        )
+    s.copy(library, excludes=['package.json', 'README.md', 'src/index.ts'])
 
 # Copy common templates
 common_templates = gcp.CommonTemplates()
-templates = common_templates.node_library()
+templates = common_templates.node_library(source_location='build/src')
 s.copy(templates)
 
+s.replace('src/v1beta1/index.ts', 'export {GrafeasClient} from \'./grafeas_client\';', '')
+s.replace('src/v1/index.ts', 'export {GrafeasClient} from \'./grafeas_client\';', '')
+s.replace('src/index.ts', 'v1beta1.GrafeasClient', 'v1beta1.GrafeasV1Beta1Client')
+
 # fix the URL of grafeas.io (this is already fixed upstream).
-s.replace('src/v1beta1/*.js',
-        'cloud.google.comgrafeas.io',
-        'grafeas.io')
+s.replace('src/v1beta1/*.ts',
+        'grafeas.io',
+        'https://grafeas.io')
 
-# perform surgery inserting the Grafeas client.
-s.replace("src/v1/container_analysis_client.js",
-r"""const path = require\('path'\);
+s.replace('tslint.json', '"extends": "gts/tslint.json"', '"extends": "gts/tslint.json", "linterOptions": {"exclude": ["src/index.ts"]}')
+# perform surgery inserting the Grafeas client. 
+# TODO: replacement not working, ts client further needs opts: ClientOptions
+s.replace("src/v1/container_analysis_client.ts",
```

Comment:
This "surgery", unfortunately, has to stay, but let's talk tomorrow and see if we can figure out another way of doing this.